### PR TITLE
[Merged by Bors] - vote on genesis

### PR DESCRIPTION
**Pull request by almogdepaz**
## Motivation
we vote on genesis only in blocks from layer one, this can cause pbase to get stuck
if we dont have enough blocks in layer 1

## Changes
blocks will vote on genesis while it is in the rage defined by hdist

## Test Plan
changed relevant tests

### CODE REVIEW

File: miner/block_builder.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func calcHdistRange(id types.LayerID, hdist types.LayerID) (bottom types.LayerID
 
 	bottom = types.LayerID(1)
 	top = id - 1
-	if id > hdist {
+	if id >= hdist {
```
Comment (barakshani) [inline miner/block_builder.go]: this is unnecessary

File: miner/block_builder.go [hunk 2]
```diff
@@ -174,7 +174,7 @@ func calcHdistRange(id types.LayerID, hdist types.LayerID) (bottom types.LayerID
 
 	bottom = types.LayerID(1)
```
Comment (barakshani) [inline miner/block_builder.go]: change to LayerID(0)

**Review by barakshani — APPROVED** [review]

**Comment by almogdepaz** [comment]
bors merge

**Comment by almogdepaz** [comment]
bors merge

**Comment by barakshani** [comment]
Not critical, but we need to remove the log "could not get result for bottom layer. adding the whole layer instead" for Genesis block

**Pull request closed without merging**